Assets/Scripts/Controller2.cs

```diff
@@ -108,20 +109,63 @@ void move(float moveHorizontal, float moveVertical)
 			direction = new Vector3 (0, -1, 0);
 		}
 
+		GameObject toInteract = objectToInteract (direction);
+		if (toInteract != null) {
+			Debug.Log (toInteract.name);
+			toInteract.GetComponent<Interact> ().interact ();
+		}
+			
+	}
+
+	private GameObject objectToInteract(Vector3 direction){
 		LayerMask layer = 1 << LayerMask.NameToLayer ("Interactive") | 1 << LayerMask.NameToLayer ("Wall");
+		List<GameObject> interactables = new List<GameObject> ();
+
+		// create raycast from -45 degrees to 45 degrees
+		for (int delta_degree = 1; delta_degree < 90; delta_degree++) {
+			Quaternion quartenion = Quaternion.AngleAxis (delta_degree-45, Vector3.forward);
+			RaycastHit2D[] hits = 
+				Physics2D.RaycastAll(this.transform.position, quartenion * direction,1.5f, layer);
+			Debug.DrawRay (this.transform.position ,quartenion * direction*1.5f, Color.green,0.1f);
+
+			// prevent wall to come between player and interactive objects
+			foreach (RaycastHit2D hit in hits) {
+				if (hit.collider != null) {
+					if (hit.collider != null) {
+						if (hit.collider.CompareTag ("impenetrable"))
+							break;
+						// Record interatables item
+						interactables.Add (hit.collider.gameObject);
+					}
+				}
+			}
 
-		RaycastHit2D[] hits = 
-			Physics2D.RaycastAll(this.transform.position, direction,1.5f, layer);
-		Debug.DrawRay (this.transform.position ,direction*1.5f, Color.green,0.2f);
+		}
+
+		if (interactables.Count == 0)
+			return null;
+
+		// most common interactable in the list is the closest one
+		return mostCommon(interactables.ToArray());
+	}
+
+	private GameObject mostCommon(GameObject[] interactables){
```

Comment:
User this instead to get the mostCommon element:

> var most = list.GroupBy(i=>i).OrderByDescending(grp=>grp.Count())
      .Select(grp=>grp.Key).First();
http://stackoverflow.com/questions/355945/find-the-most-occurring-number-in-a-listint

Im surprised that the C# list doesn't provide a most common element function by default. But then again Im blessed to work with scala at work.